linkup/src/lib.rs

```diff
@@ -61,62 +63,67 @@ pub fn get_additional_headers(
         let flags_hex = hex::encode(flags);
 
         let traceparent = format!("{}-{}-{}-{}", version_hex, trace_hex, parent_hex, flags_hex);
-        additional_headers.insert("traceparent".to_string(), traceparent);
+        additional_headers.insert(HeaderName::TraceParent, traceparent);
     }
 
-    let tracestate = headers.get("tracestate");
+    let tracestate = headers.get(HeaderName::TraceState);
     let linkup_session = format!("linkup-session={}", session_name,);
     match tracestate {
         Some(ts) if !ts.contains(&linkup_session) => {
             let new_tracestate = format!("{},{}", ts, linkup_session);
-            additional_headers.insert("tracestate".to_string(), new_tracestate);
+            additional_headers.insert(HeaderName::TraceState, new_tracestate);
         }
         None => {
             let new_tracestate = linkup_session;
-            additional_headers.insert("tracestate".to_string(), new_tracestate);
+            additional_headers.insert(HeaderName::TraceState, new_tracestate);
         }
         _ => {}
     }
 
-    if !headers.contains_key("linkup-destination") {
-        additional_headers.insert(
-            "linkup-destination".to_string(),
-            destination_service_name.to_string(),
-        );
+    if !headers.contains_key(HeaderName::LinkupDestination) {
+        additional_headers.insert(HeaderName::LinkupDestination, &target_service.name);
     }
 
-    if !headers.contains_key("x-forwarded-host") {
+    if !headers.contains_key(HeaderName::ForwardedHost) {
         additional_headers.insert(
-            "x-forwarded-host".to_string(),
-            get_target_domain(&url, session_name),
+            HeaderName::ForwardedHost,
+            get_target_domain(url, session_name),
         );
     }
 
+    additional_headers.insert("host", Url::parse(&target_service.url).unwrap());
```

Comment:
This is where I wonder if we need an if statement of some kind or not...

Reply:
Moved back to the linkup-cli on f4cf4b74054bbc7a54801032127ab4b437372ec6